src/System.IO.Ports/src/System/IO/Ports/SerialPort.cs

```diff
@@ -1204,12 +1204,18 @@ private void CatchReceivedEvents(object src, SerialDataReceivedEventArgs e)
                     }
                     catch
                     {
-                        // Ignore and continue. SerialPort might have been closed already! 
+                        // Ignore and continue. SerialPort might have been closed already!
                     }
                     finally
                     {
+                        // ISSUE: This should be fired only when it wasn't already fired for the total number of bytes available
+                        //        Similarly as done in SerialStream.Linux (IOLoop)
+                        //        I.e: Let _receivedBytesThreshold be 8 - when we get an event when 7 bytes are available
+                        //        BytesToRead can change while we run this event and thus
+                        //        we virtually can get 2 events when 8th byte arrives
+                        //        I.e. we might want to add total bytes available as internal field in the args event
```

Comment:
I'm currently tracking all of the issues in the PR description, once this will be stable enough I'll file them so that I don't do much churn with creating issues.